packages/eas-cli/src/update/queries.ts

```diff
@@ -0,0 +1,93 @@
+import chalk from 'chalk';
+import CliTable from 'cli-table3';
+
+import { PaginatedQueryFlags } from '../commandUtils/flagHelpers';
+import { BranchUpdateObject, UpdateQuery } from '../graphql/queries/UpdateQuery';
+import Log from '../log';
+import formatFields from '../utils/formatFields';
+import { printJsonOnlyOutput } from '../utils/json';
+import {
+  PaginatedQueryPromptOptions,
+  PaginatedQueryPromptType,
+  PaginatedQueryResponse,
+  performPaginatedQueryAsync,
+} from '../utils/queries';
+import { UPDATE_COLUMNS, formatUpdate, getUpdateGroupsWithPlatforms } from './utils';
+
+export const UPDATES_LIMIT = 50;
+
+export async function queryUpdatesOnBranchByNameAsync(
+  projectId: string,
+  branchName: string,
+  flags: PaginatedQueryFlags
+): Promise<void> {
+  let updateBranch: { name: string; id: string };
+  const queryUpdatesGroupsForBranchAsync = async (
+    pageSize: number,
+    offset: number
+  ): Promise<PaginatedQueryResponse<BranchUpdateObject>> => {
+    const { app } = await UpdateQuery.viewBranchAsync({
+      appId: projectId,
+      name: branchName,
+      limit: pageSize,
+      offset,
+    });
+
+    const UpdateBranch = app?.byId.updateBranchByName;
+    if (!UpdateBranch) {
+      throw new Error(`Could not find branch "${branchName}"`);
+    }
+
+    updateBranch = { id: UpdateBranch.id, name: UpdateBranch.name };
+    return {
+      queryResponse: UpdateBranch.updates,
+      queryResponseRawLength: UpdateBranch.updates.length,
+    };
+  };
```

Comment:
local functions (nested functions) are a feature of the javascript language and provide scoped functions specific to the execution context. This lets us define behavior that has access to the execution context's variables, co-locate function declaration with usage, make it clear that this is not a shared function to be used in other locations in the code-base, and label a block of code to help developers understand what it's doing. We have examples of them being used in other parts of the code-base (`ora/ora`, `build/runLocalBuildAsync`, `deviceUtils/chooseDevicesAsync`). Is this a nit? or should we add some linting and remove other usages from the code-base?

Reply:
I think using local functions is fine but shouldn't be overused. In this particular case, the parent function is really long because it has nested functions (~100 LOC). 

If you're not familiar with code, it's much easier to read functions (and understand them) that look like this
```
function complexFunction() {
  const x = doA();
  const y = parseX(x);
  printY(y);
} 
```
rather than sth like this

```
function complexFunction() {
  const x = () => {
    // 150 lines of code
    // ...
    // even more lines of code 
  };
  const y = () => {
    // 2137 lines of code
    // ...
    // even more lines of code 
  };
  printY(y);
} 
```

Reply:
I think using local functions is fine but shouldn't be overused. In this particular case, the parent function is really long because it has nested functions (~100 LOC). 

If you're not familiar with code, it's much easier to read functions (and understand them) that look like this
```
function complexFunction() {
  const x = doA();
  const y = parseX(x);
  printY(y);
} 
```
rather than sth like this
```
function complexFunction() {
  const x = (() => {
    // 150 lines of code
    // ...
    // even more lines of code 
  })();
  const y = (() => {
    // 2137 lines of code
    // ...

    // out of nowhere, this function uses `x` from the parent function (hard to refactor and test!)

    // even more lines of code 
  })();
  (() => {
    // 150 lines of code
    // ...

    //  use `y` from parent function

    // even more lines of code 
  });
} 
```